core/src/main/java/com/datastax/oss/driver/internal/core/config/typesafe/DefaultDriverConfigLoader.java

```diff
@@ -140,47 +155,59 @@ private SingleThreaded(InternalDriverContext context) {
               .getDefaultProfile()
               .getDuration(DefaultDriverOption.CONFIG_RELOAD_INTERVAL);
 
-      forceLoadListenerKey =
-          this.eventBus.register(
-              ForceReloadConfigEvent.class, e -> RunOrSchedule.on(adminExecutor, this::reload));
-
-      RunOrSchedule.on(adminExecutor, this::scheduleReloadTask);
+      RunOrSchedule.on(adminExecutor, this::schedulePeriodicReload);
     }
 
-    private void scheduleReloadTask() {
+    private void schedulePeriodicReload() {
       assert adminExecutor.inEventLoop();
       // Cancel any previously running task
-      if (reloadFuture != null) {
-        reloadFuture.cancel(false);
+      if (periodicTaskHandle != null) {
+        periodicTaskHandle.cancel(false);
       }
       if (reloadInterval.isZero()) {
         LOG.debug("[{}] Reload interval is 0, disabling periodic reloading", logPrefix);
       } else {
         LOG.debug("[{}] Scheduling periodic reloading with interval {}", logPrefix, reloadInterval);
-        reloadFuture =
+        periodicTaskHandle =
             adminExecutor.scheduleAtFixedRate(
-                this::reload,
+                () -> reload(null),
```

Comment:
I don't understand what is an intention of passing `null` here. Is it for triggering reload without returning value to the user? If so I am suggesting to create some constant marker with `null` assigned to it. Like `TRIGGER_RELOAD_WITHOUT_RETURNIG_RESULT` and use that constant in the code. It will make the intention clean and signal that null here has a special meaning.

Reply:
The method is now used in two cases:
* manual reload (added by this PR): we want to let the client application know when it's done, hence the `CompletionStage`.
* periodic reload (what was already there): happens in the background so nobody's listening directly, no need to create an unnecessary object.

So yeah `null` is just a simple way to say "no stage to complete", this is internal code so we don't need to be super fancy.
On the other hand this is not on the hot path so I wouldn't mind using `Optional`, would that make the code easier to read?

Reply:
I don't think that Optional will make it cleaner. I am suggesting just adding constant and assigning `null` or `Optional` to it will be fine

Reply:
I added a shortcut to make the method reference more explicit, and a bit of documentation.

As for extracting a constant,  I don't think it improves things here because what we want to express is the absence of a reference, and this is precisely what `null` means (whereas if it's a constant it's not clear that you should never dereference it). Besides this is internal code, and contained to this short class, so it's pretty trivial.